include/caffe/layer.hpp

```diff
@@ -413,7 +413,7 @@ template <typename Dtype>
 inline Dtype Layer<Dtype>::Forward(const vector<Blob<Dtype>*>& bottom,
     const vector<Blob<Dtype>*>& top) {
   Dtype loss = 0;
-  Reshape(bottom, top);
+  // Reshape(bottom, top);
```

Comment:
@cyberfire You should you leave your PR open. The core developers will take a look at it.